ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,694 @@
+		replacement.prev = vertex_a.prev
+		replacement.next = vertex_b.next
+
+		vertex_a.invalidate()
+		vertex_b.invalidate()
+
+		self._len -= 1
+		return replacement
+
+	def __str__(self):
+		return 'LAV {}'.format(id(self))
+
+	def __repr__(self):
+		return '{} = {}'.format(str(self), [vertex for vertex in self])
+
+	def __len__(self):
+		return self._len
+
+	def __iter__(self):
+		cur = self.head
+		while True:
+			yield cur
+			cur = cur.next
+			if cur == self.head:
+				raise StopIteration
+
+	def _show(self):
+		cur = self.head
+		while True:
+			print(cur.__repr__())
+			cur = cur.next
+			if cur == self.head:
+				break
+
+class _EventQueue:
+	def __init__(self):
+		self.__data = []
+
+	def put(self, item):
+		if item is not None:
+			heapq.heappush(self.__data, item)
+
+	def put_all(self, iterable):
+		for item in iterable:
+			heapq.heappush(self.__data, item)
+
+	def get(self):
+		return heapq.heappop(self.__data)
+
+	def empty(self):
+		return len(self.__data)==0
+
+	def peek(self):
+		return self.__data[0]
+
+	def show(self):
+		for item in self.__data:
+			print(item)
+
+
+def skeletonize(polygon, holes=None):
+	"""
+	Compute the straight skeleton of a polygon.
+
+	The polygon should be given as a list of vertices in counter-clockwise order.
+	Holes is a list of the contours of the holes, the vertices of which should be in
+	clockwise order.
+
+	Returns the straight skeleton as a list of "subtrees", which are in the form of
+	(source, height, sinks), where source is the highest points, height is its height,
+	and sinks are the point connected to the source.
+	"""
+
+	#HACK: But may be least bad way to add this functionality w/o contaminating
+	#rest of LB point geometry.
+	# Add polyskel-specific point2d magic method to Point2D used by heapq
+	Point2D.__lt__ = _polyskel_point2d_lt
+
+	# Current code actually works on cw and ccw order for polygons and holes,
+	# respectively. So reverse vertice order for both inputs
+	polygon = polygon[::-1]
+	if holes is not None:
+		holes = [hole[::-1] for hole in holes]
+
+	slav = _SLAV(polygon, holes)
+	output = []
+	prioque = _EventQueue()
+
+	for lav in slav:
+		for vertex in lav:
+			v = vertex.next_event()
+			prioque.put(v)
+
+	while not (prioque.empty() or slav.empty()):
+		log.debug('SLAV is %s', [repr(lav) for lav in slav])
+		i = prioque.get()
+		if isinstance(i, _EdgeEvent):
+			if not i.vertex_a.is_valid or not i.vertex_b.is_valid:
+				log.info('%.2f Discarded outdated edge event %s', i.distance, i)
+				continue
+
+			(arc, events) = slav.handle_edge_event(i)
+		elif isinstance(i, _SplitEvent):
+			if not i.vertex.is_valid:
+				log.info('%.2f Discarded outdated split event %s', i.distance, i)
+				continue
+			(arc, events) = slav.handle_split_event(i)
+
+		prioque.put_all(events)
+
+		if arc is not None:
+			output.append(arc)
+			for sink in arc.sinks:
+				_debug.line((arc.source.x, arc.source.y, sink.x, sink.y), fill='red')
+
+			_debug.show()
+	return output
```

Comment:
Hmm, I did set tab to be 4 spaces in Pycharm. And I've counted the spaces here and there to confirm it's working Where do you see the 6/8 spaces? It might just be deeply nested loops/conditions?

Reply:
It's only in the `polyskel.py` file. But the tabs are jumping all over the place there. It might be clearer to you if you take one line that's indented from the previous one and count the number of characters you see in the indent.